enterprise/server/workflow/service/service.go

```diff
@@ -1143,37 +1141,21 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 		return nil, err
 	}
 
-	timeout := *workflowsDefaultTimeout
+	timeout := *ci_runner_util.CIRunnerDefaultTimeout
 	if workflowAction.Timeout != nil {
 		timeout = *workflowAction.Timeout
 	}
 
-	// NOTE: By default, the executor is responsible for ensuring the
-	// buildbuddy_ci_runner binary exists at the workspace root when it sees
-	// the `workflow-id` platform property.
-	inputRootDigest, err := digest.ComputeForMessage(&repb.Directory{}, repb.DigestFunction_SHA256)
-	if err != nil {
-		return nil, err
-	}
-	if (os == "" || os == platform.LinuxOperatingSystemName) && (workflowAction.Arch == "" || workflowAction.Arch == "amd64") {
-		// Because the apps are built for linux/amd64, in these cases we can have the apps
-		// upload the ci_runner binary to the cache to ensure the executors are using
-		// the most up-to-date version of the binary
-		runnerBinDigest, err := cachetools.UploadBlobToCAS(ctx, ws.env.GetByteStreamClient(), instanceName, repb.DigestFunction_SHA256, ci_runner_bundle.CiRunnerBytes)
+	var inputRootDigest *repb.Digest
+	if ci_runner_util.CanInitFromCache(os, workflowAction.Arch) {
```

Comment:
it seems like this `if/else` block could be moved into the function that uploads the input root dir, to simplify both callsites?